### PR TITLE
DS-214 Listing Teaser Component

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-214

## Summary

Adds the Listing Teaser component, an interactive UI element that summarizes a particular listing.

## Details

1. The sting Teaser component only has twig templates, no JS or web components.
2. It can be used to display specific information of a particular listing, article, or thread. It is commonly used on listing pages.
3. It accommodates use cases in both Academy and Collaboration Center.

### Example usage

```twig
{% set icon_questions %}
  {% include '@bolt-components-icon/icon.twig' with {
    name: 'questions',
    size: 'xlarge',
    color: 'teal',
  } only %}
{% endset %}
{% include '@bolt-components-listing-teaser/listing-teaser.twig' with {
  signifier: icon_questions,
  headline: {
    text: 'This is a listing',
    tag: 'h3',
    size: 'xlarge',
    link_attributes: {
      href: 'https://pega.com'
    },
  },
  meta_items: [
    'Posted 8 hours 15 minutes ago',
    'Last activity: 2 minutes ago',
  ],
  status: {
    solved: true,
    locked: true,
    replies: '33',
    views: '6k',
  },
  action_menu: action_menu,  // Pass Menu component here
  share_menu: share_menu,  // Pass Share component here with display set to "menu"
  more_info: 'This is the more information. Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus',
  warning: 'Warning! This is a warning message.',
} only %}
```

## How to test

Run the branch locally, do the following:

1. Check all the docs under Listing Teaser in PL.
2. Check the twig template setup.
3. Check the schema.
4. Create a dummy page and try to use the component as instructed in the docs.

### CODE REVIEW

File: packages/components/bolt-post/src/post.twig [hunk 1]
```diff
@@ -0,0 +1,189 @@
+{# Schema Validation #}
+{% set schema = bolt.data.components['@bolt-components-post'].schema %}
+{% if enable_json_schema_validation %}
+  {{ validate_data_schema(schema, _self)|raw }}
+{% endif %}
+
+{# Variables #}
+{% set this = init(schema) %}
+{% set attributes = create_attribute(attributes|default({})) %}
+{% set _post_id = _post_id|default(bolt.data.config.env == 'test' ? 'js-test-target' : 'js-bolt-post-' ~ random()) %}
```
Comment (danielamorse) [inline packages/components/bolt-post/src/post.twig:10]: It looks like `_post_id` is only set internally, i.e. you don't ever pass `_post_id` into the template. Is the `default()` necessary?

File: packages/components/bolt-post/src/post.twig [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+<article {{ attributes.addClass(classes) }}>
+  <div class="c-bolt-post__flag">
+    {% if signifier %}
+      <div class="c-bolt-post__flag-signifier" aria-hidden="true">
+        {{ signifier }}
+      </div>
+    {% endif %}
+    <div class="c-bolt-post__flag-heading">
+      {% set _headline_link_attributes = headline.link_attributes %}
+      {% if headline.link_attributes %}
```
Comment (danielamorse) [inline packages/components/bolt-post/src/post.twig:27]: If you're setting `_headline_link_attributes` variable, might as well use it on this line too.

**Review by danielamorse — COMMENTED** [review]
Nice, this will be great to have on Academy. Code review and xbrowser testing are all good. I left a couple comments inline.

When status items wrap do you still want the left border?
image

Currently this doesn't give users a way to modify the content of the actions (reply, favorite, etc). I assume that's something they'll need in order to integrate it with Drupal. Any thoughts?

File: packages/components/bolt-post/src/post.twig (hunk 2 above)
Comment (mikemai2awesome) [reply packages/components/bolt-post/src/post.twig:27]: that was my thought, but it didn't work as expected. we can look at it together.

File: packages/components/bolt-post/src/post.twig (hunk 1 above)
Comment (mikemai2awesome) [reply packages/components/bolt-post/src/post.twig:10]: oh right, i should just set the var straight.

File: packages/components/bolt-post/src/post.twig (hunk 2 above)
Comment (mikemai2awesome) [reply packages/components/bolt-post/src/post.twig:26]: now it works. not sure why it was not working before. previously it was rendering all the props as attributes, things like `reversed_underline` and `expand_click_target` were added as attributes. weird.

**Comment by mikemai2awesome** [comment]
@danielamorse The border is fine. It lines up the content, without it then it looks like the items are indented.

Good catch on the popover actions, I need to pick your brains on that.

**Comment by mikemai2awesome** [comment]
@danielamorse Action menu and share menu props are updated after our discussion.

**Review by danielamorse — APPROVED** [review]
Updates look good, approved.

This has me thinking about how we use nested props. Nested props make sense here -- you do want all headline props to be under `headline` once there are enough of them. But we're not leveraging `init()` when we do that, and then we end up with a mix of conventions.

I'm good merging this as is, but I'd like to discuss this use case in huddle tomorrow.

**Comment by deepakqkumar** [comment]
@mikemai2awesome - on pages where we have unfavorite or unsubscribe icons before the node type icon, can you help understand how we should insert those action icons with this new component?
image

**Comment by mikemai2awesome** [comment]
Hi @deepakqkumar, as we discussed in Slack, all actions are consolidated into the action menu popover.
Screen Shot 2021-02-17 at 10 19 06 AM

**Review by remydenton — CHANGES_REQUESTED** [review]
I think the main thing to do before merging is change the component name to 'listing teaser'.  I added one other optional comment.

File: packages/components/bolt-post/post.schema.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      description:
+        'Set a visual signifier for the post. An icon or decorative image usually goes here.',
+    },
+    headline: {
+      type: 'object',
+      description: 'Render the headline of the post.',
+      properties: {
+        text: {
+          type: 'string',
+          description: 'Set the text of the headline.',
```
[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply packages/components/bolt-post/post.schema.js]: Updated.

**Comment by mikemai2awesome** [comment]
@remydenton Renamed and updated schema as instructed.

**Review by remydenton — APPROVED** [review]
Updates look great, thanks @mikemai2awesome.

**Pull request merged**